### ISSUE TITLE
Allow Viewer to Select Stream Quality Preference

### ISSUE THREAD

**Issue by GTXCrusader** [owncast/owncast#907]
Currently if there are multiple transcoding qualities being transmitted, the quality is automatically selected based on the viewer's network connectivity. We could not only allow viewers to continue using the automatic feature, but also allow them to select from multiple qualities if they are there.

This can be useful for viewers who are monitoring data usage (such as cellular data) but are on high-speed connections and would like to lower the quality to save on bandwidth.

This can also be useful to troubleshoot and verify specific transcoding settings, as well as to "lock" a specific quality. If a viewer selects a quality that is above what their network can handle, then similar to how a service like Twitch will handle it, the stream will buffer until the user changes to either "Auto" or a quality more fitting for their connection.

The quality names/settings could be set in the server configuration so it can be named according to the options (eg: "Auto, 1080p, 720p, 480p" or "Auto, 7mbps, 4mpbs, 1mbps"). Perhaps this could be done by creating a new field in the "Edit Video Variant Details" that would be a simple Title, and that Title is what would be used to name the qualities.

A possible more advanced means of creating titles and to prevent abuse would be to take the standard options of determining stream quality (Resolution, Bitrate, Framerate), and make them selectable options only based on the selections under the Edit Video Variant Details page. The admin could then select what qualities they would want to show as selections, and it could be simple ("1080p") or complex ("1080p 60fps 5000kbps").

To summarize:

- Add selectable qualities to the web player for users to select (if multiple qualities are available).
- Allow options to have their displays customized by the Admin (either full freedom or selectable options).
- If "Video Passthrough" is Selected as a Variant, quality name will simply be "Source".
- If only a single quality is available, default the selection to "Source" or "Auto".
- Retain the automatic feature of the stream via an "Auto" option.

**Comment by GTXCrusader** [comment]
Ah ha! I had browsed through and did not catch that. Apologies, and yay!
